### PR TITLE
[map vis tool] support manual geojson distances

### PR DESCRIPTION
**Pull request by chejennifer**
- update map tile to be able to calculate geojson for specific place types (this will propagate to map vis tool)

### CODE REVIEW

File: static/js/components/tiles/map_tile.tsx [hunk 1]
```diff
@@ -430,10 +435,29 @@ function rawToChart(
...
     const metadataMap = rawData.placeStat.facets || {};
     const placeStat = rawData.placeStat.data[statVarSpec.statVar] || {};
+    // Get the list of child places from either the geojson data or the
+    // placeStat data
+    const childPlaces = !_.isEmpty(rawData.geoJson.features)
+      ? rawData.geoJson.features.map((feature) => feature.properties.geoDcid)
+      : Object.keys(placeStat);
+    let geoJson = rawData.geoJson;
+    if (
+      rawData.enclosedPlaceType in MANUAL_GEOJSON_DISTANCES &&
```
Comment (juliawu) [inline static/js/components/tiles/map_tile.tsx:445]: Do we need to add an entry for `GeoGridPlace_1Deg` to `MANUAL_GEOJSON_DISTANCES` for 1 degree grids to work?

Comment (chejennifer) [reply static/js/components/tiles/map_tile.tsx:445]: it's already there which is why 1 degree grids work on the old map tool

**Review by shifucun — COMMENTED** [review]
wonder how ipcc data works now for the new vis tool?

**Comment by chejennifer** [comment]
> wonder how ipcc data works now for the new vis tool?

Is your question:

a. whether or not IPCC works now for the new vis tool
b. why does IPCC work now for the new vis tool?

Also, by IPCC what is the place you are showing?

**Comment by shifucun** [comment]
> > wonder how ipcc data works now for the new vis tool?
> 
> Is your question:
> 
> a. whether or not IPCC works now for the new vis tool b. why does IPCC work now for the new vis tool?
> 
> Also, by IPCC what is the place you are showing?

I mean, how does this page work now: https://datacommons.org/tools/visualization#visType%3Dmap%26place%3Dcountry%2FIND%26placeType%3DIPCCPlace_50%26sv%3D%7B%22dcid%22%3A%22DifferenceRelativeToBaseDate2006_Max_Temperature_RCP45%22%7D

**Comment by chejennifer** [comment]
> > > wonder how ipcc data works now for the new vis tool?
> > 
> > 
> > Is your question:
> > a. whether or not IPCC works now for the new vis tool b. why does IPCC work now for the new vis tool?
> > Also, by IPCC what is the place you are showing?
> 
> I mean, how does this page work now: https://datacommons.org/tools/visualization#visType%3Dmap%26place%3Dcountry%2FIND%26placeType%3DIPCCPlace_50%26sv%3D%7B%22dcid%22%3A%22DifferenceRelativeToBaseDate2006_Max_Temperature_RCP45%22%7D

it works because we have the geojsons: https://datacommons.org/api/choropleth/geojson?placeDcid=country%2FIND&placeType=IPCCPlace_50

**Comment by shifucun** [comment]
> > > > wonder how ipcc data works now for the new vis tool?
> > > 
> > > 
> > > Is your question:
> > > a. whether or not IPCC works now for the new vis tool b. why does IPCC work now for the new vis tool?
> > > Also, by IPCC what is the place you are showing?
> > 
> > 
> > I mean, how does this page work now: https://datacommons.org/tools/visualization#visType%3Dmap%26place%3Dcountry%2FIND%26placeType%3DIPCCPlace_50%26sv%3D%7B%22dcid%22%3A%22DifferenceRelativeToBaseDate2006_Max_Temperature_RCP45%22%7D
> 
> it works because we have the geojsons: https://datacommons.org/api/choropleth/geojson?placeDcid=country%2FIND&placeType=IPCCPlace_50

I see, so even though IPCCPlace_50 is in MANUAL_GEOJSON_DISTANCES, but we have computed the geojson in cache. @pradh is this true for all IPCCPlace_50 node?

**Comment by chejennifer** [comment]
thanks for the review!

**Comment by pradh** [comment]
> > > > > wonder how ipcc data works now for the new vis tool?
> > > > 
> > > > 
> > > > Is your question:
> > > > a. whether or not IPCC works now for the new vis tool b. why does IPCC work now for the new vis tool?
> > > > Also, by IPCC what is the place you are showing?
> > > 
> > > 
> > > I mean, how does this page work now: https://datacommons.org/tools/visualization#visType%3Dmap%26place%3Dcountry%2FIND%26placeType%3DIPCCPlace_50%26sv%3D%7B%22dcid%22%3A%22DifferenceRelativeToBaseDate2006_Max_Temperature_RCP45%22%7D
> > 
> > 
> > it works because we have the geojsons: https://datacommons.org/api/choropleth/geojson?placeDcid=country%2FIND&placeType=IPCCPlace_50
> 
> I see, so even though IPCCPlace_50 is in MANUAL_GEOJSON_DISTANCES, but we have computed the geojson in cache. @pradh is this true for all IPCCPlace_50 node?

Ah, I suspect that could be because we started IPCCPlace_50 with fixed sized grids, and then migrated it to not-always-grids that might take the shape of country boundaries.

**Comment by shifucun** [comment]
> > > > > > wonder how ipcc data works now for the new vis tool?
> > > > > 
> > > > > 
> > > > > Is your question:
> > > > > a. whether or not IPCC works now for the new vis tool b. why does IPCC work now for the new vis tool?
> > > > > Also, by IPCC what is the place you are showing?
> > > > 
> > > > 
> > > > I mean, how does this page work now: https://datacommons.org/tools/visualization#visType%3Dmap%26place%3Dcountry%2FIND%26placeType%3DIPCCPlace_50%26sv%3D%7B%22dcid%22%3A%22DifferenceRelativeToBaseDate2006_Max_Temperature_RCP45%22%7D
> > > 
> > > 
> > > it works because we have the geojsons: https://datacommons.org/api/choropleth/geojson?placeDcid=country%2FIND&placeType=IPCCPlace_50
> > 
> > 
> > I see, so even though IPCCPlace_50 is in MANUAL_GEOJSON_DISTANCES, but we have computed the geojson in cache. @pradh is this true for all IPCCPlace_50 node?
> 
> Ah, I suspect that could be because we started IPCCPlace_50 with fixed sized grids, and then migrated it to not-always-grids that might take the shape of country boundaries.

Should we remove it from MANUAL_GEOJSON_DISTANCES?

**Pull request merged**